python_client/kubetorch/cli.py

```diff
@@ -1267,14 +1270,32 @@ def kt_ssh(
         # Get and validate pods
         pods = validate_pods_exist(name, namespace, core_api)
 
-        sorted_by_time = sorted(pods, key=lambda pod: pod.metadata.creation_timestamp)
-
         # case when the user provides a specific pod to ssh into
         if pod:
+            sorted_by_time = sorted(pods, key=lambda pod: pod.metadata.creation_timestamp)
             pod_name = validate_provided_pod(service_name=name, provided_pod=pod, service_pods=sorted_by_time)
```

Comment:
if we're just validating the given pod, we wouldn't need to sort the list (and same for in compute)
```suggestion
            pod_name = validate_provided_pod(service_name=name, provided_pod=pod, service_pods=pods)
```